src/components/pre-playback-play-overlay/pre-playback-play-overlay.js

```diff
@@ -94,25 +94,14 @@ class PrePlaybackPlayOverlay extends BaseComponent {
    * @memberof PrePlaybackPlayOverlay
    */
   handleClick(): void {
-    // TODO: The promise handling should be in the play API of the player.
-    new Promise((resolve, reject) => {
-      try {
-        if (this.player.config.playback.preload === "auto" && !this.player.config.plugins.ima) {
-          this.player.ready().then(resolve);
-        } else {
-          resolve();
-        }
-      } catch (e) {
-        reject(e);
-      }
-    }).then(() => {
+    try {
```

Comment:
@odedhutzler The try catch was there to protect against the nested player accessor I think, please check if still required.
Otherwise, LGTM.